src/backend/optimizer/plan/createplan.c

```diff
@@ -782,17 +782,6 @@ create_join_plan(PlannerInfo *root, JoinPath *best_path)
 	if (partition_selector_created)
 		((Join *) plan)->prefetch_inner = true;
 
-	/* CDB: if the join's locus is bottleneck which means the
-	 * join gang only contains one process, so there is no
-	 * risk for motion deadlock.
-	 */
-	if (CdbPathLocus_IsBottleneck(best_path->path.locus))
-	{
-		((Join *) plan)->prefetch_inner = false;
-		((Join *) plan)->prefetch_joinqual = false;
```

Comment:
Something I can’t find, where are the prefetch_joinqual and prefetch_qual fields used?

Reply:
Maybe they should be removed altogether, since they are no longer used after e0cca545510322805731d5eb723f512df6b25f29?

Reply:
@KnightMurloc,  Change it back